code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceCreate.java

```diff
@@ -153,13 +163,37 @@ protected Volume getRoot(Instance instance, List<Volume> volumes) {
     }
 
     protected Set<Long> createNics(Instance instance, List<Nic> nics, Map<String, Object> data) {
-        List<Long> networkIds = DataUtils.getFieldList(instance.getData(), InstanceConstants.FIELD_NETWORK_IDS, Long.class);
+        List<Long> networkIds = populateNetworks(instance);
         List<Long> subnetIds = DataUtils.getFieldList(instance.getData(), InstanceConstants.FIELD_SUBNET_IDS, Long.class);
         List<Long> vnetIds = DataUtils.getFieldList(instance.getData(), InstanceConstants.FIELD_VNET_IDS, Long.class);
-
         return createNicsFromIds(instance, nics, data, networkIds, subnetIds, vnetIds);
     }
 
+    protected List<Long> populateNetworks(Instance instance) {
+        List<Long> networkIds = DataUtils.getFieldList(instance.getData(), InstanceConstants.FIELD_NETWORK_IDS,
+                Long.class);
+        Label dnsLabel = labelsDao.getLabelForInstance(instance.getId(), SystemLabels.LABEL_USE_RANCHER_DNS);
```

Comment:
Why do you refer to the database and not the `Labels` field on the container?  During create I would almost assume the labels table wouldn't be populated.  Regardless, the Labels field on the container should be the source of truth and the other db tables are just a normalized form to make some queries better.